Sources/Web5/Crypto/JOSE/JWT.swift

```diff
@@ -72,6 +79,63 @@ public struct JWT {
             case issuedAt = "iat"
             case jwtID = "jti"
         }
+        
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+
+            // Decode the known properties
+            issuer = try container.decodeIfPresent(String.self, forKey: .issuer)
+            subject = try container.decodeIfPresent(String.self, forKey: .subject)
+            audience = try container.decodeIfPresent(String.self, forKey: .audience)
+            expiration = try container.decodeIfPresent(Int.self, forKey: .expiration)
+            notBefore = try container.decodeIfPresent(Int.self, forKey: .notBefore)
+            issuedAt = try container.decodeIfPresent(Int.self, forKey: .issuedAt)
+            jwtID = try container.decodeIfPresent(String.self, forKey: .jwtID)
+
+            // Initialize the miscellaneous dictionary
+            var misc = [String: AnyCodable]()
+
+            // Extract all rawValues from CodingKeys for comparison
+            let knownKeysRawValues = CodingKeys.allCases.map { $0.rawValue }
+
+            // Dynamically decode keys not in CodingKeys
+            let dynamicContainer = try decoder.container(keyedBy: DynamicCodingKey.self)
+            for key in dynamicContainer.allKeys {
+                // Convert DynamicCodingKey to String
+                let keyString = key.stringValue
+                
+                // Skip keys that are part of the known CodingKeys
+                if !knownKeysRawValues.contains(keyString) {
+                    if let value = try? dynamicContainer.decode(AnyCodable.self, forKey: key) {
+                        misc[keyString] = value
+                    }
+                }         
+            }
+
+            miscellaneous = misc.isEmpty ? nil : misc
+        }
+        
+        public func encode(to encoder: Encoder) throws {
+            var container = encoder.container(keyedBy: CodingKeys.self)
+
+            // Encode known properties
+            try container.encodeIfPresent(issuer, forKey: .issuer)
+            try container.encodeIfPresent(subject, forKey: .subject)
+            try container.encodeIfPresent(audience, forKey: .audience)
+            try container.encodeIfPresent(expiration, forKey: .expiration)
+            try container.encodeIfPresent(notBefore, forKey: .notBefore)
+            try container.encodeIfPresent(issuedAt, forKey: .issuedAt)
+            try container.encodeIfPresent(jwtID, forKey: .jwtID)
+
+            // Dynamically encode the miscellaneous properties at the top level
```

Comment:
> at the top level

So this means the claims body won't look like this, right?

```json
{
  // existing well-defined claims properties
  "misc": {
    "something": "else"
  }
}
```

and instead will look like this, right?

```json
{
  // existing well-defined claims properties
  "something": "else"
}
```

(idk Swift so apologies in advance)